rustls/src/crypto/ring/ticketer.rs

```diff
@@ -93,14 +102,26 @@ impl ProducesTickets for AeadTicketer {
             .fill(&mut nonce_buf)
             .ok()?;
         let nonce = aead::Nonce::assume_unique_for_key(nonce_buf);
-        let aad = aead::Aad::empty();
-
-        let mut ciphertext =
-            Vec::with_capacity(nonce_buf.len() + message.len() + self.key.algorithm().tag_len());
+        let aad = aead::Aad::from(self.key_name);
```

Comment:
I'm a little confused by the notion of sticking the key name both in the AAD and in the ciphertext. If we stick it in the ciphertext, what's the value of also putting it in the AAD?

Reply:
I mentioned this a bit in the commit message: doing both is unnecessary, but also not harmful.

Reply:
I'm probably being stupid or lack your cryptography expertise, but to me this doesn't really follow from your commit message. Anyway -- if doing both is unnecessary, why do it? Also I think this merits a comment in the code where it provides context that isn't readily available from the commit message.

Reply:
Added a comment and some references, especially one to https://eprint.iacr.org/2022/1680.pdf where figure 6 matches what is going on here pretty well.